cumulusci/tasks/salesforce/Deploy.py

```diff
@@ -42,29 +41,56 @@ class Deploy(BaseSalesforceMetadataApiTask):
         },
     }
 
+    def _get_api(self, path=None):
+        if not path:
+            path = self.task_config.options__path
+
+        package_zip = self._get_package_zip(path)
+        self.logger.info("Payload size: {} bytes".format(len(package_zip)))
+        return self.api_class(self, package_zip, purge_on_delete=False)
+
+    def _include_directory(self, root_parts):
+        # include the root directory, all non-lwc directories and sub-directories, and lwc component directories
+        return len(root_parts) == 0 or root_parts[0] != "lwc" or len(root_parts) == 2
+
+    def _include_file(self, root_parts, f):
+        if len(root_parts) == 2 and root_parts[0] == "lwc":
+            # is file of lwc component directory
+            lower_f = f.lower()
+            return (
+                lower_f.endswith(".js")
+                or lower_f.endswith(".js-meta.xml")
+                or lower_f.endswith(".html")
+                or lower_f.endswith(".css")
+                or lower_f.endswith(".svg")
+            )
+        return True
+
+    def _get_files_to_package(self, path):
+        files_to_pacakge = []
+        for root, dirs, files in os.walk("."):
+            root_parts = root.split(os.sep)[1:]
+            if self._include_directory(root_parts):
+                for f in files:
+                    if self._include_file(root_parts, f):
+                        files_to_pacakge.append(os.path.join(root, f))
```

Comment:
I like the factoring out of `_get_files_to_package`. Since the method is only used as an iterator, I would suggest turning it into a generator by using the `yield` statement. Just turn this line into `yield os.path.join(root, f)` and remove the first and last lines of the method. This has the benefit of only holding one filename in memory at a time rather than constructing the full list before returning it.

Reply:
Done.

Wow, that's a neat concept that's new to me.  I feel so surprised that the the test method just passed as it's currently written, but I'm guess that when a generator is called outside a `for in`, the system collects all values.

Reply:
@spelak-salesforce it passed because `set()` iterates over the argument you pass it to construct the set.